src/algebra/order/ring.lean

```diff
@@ -790,41 +696,7 @@ class ordered_ring (α : Type u) extends ring α, ordered_add_comm_group α :=
 (mul_pos     : ∀ a b : α, 0 < a → 0 < b → 0 < a * b)
```

Comment:
Should this be changed as well? Do we need `strict_ordered_ring` and `ordered_ring`?

Reply:
Indeed it should, as I explained [here](https://github.com/leanprover-community/mathlib/pull/16172#issuecomment-1242318749). But I would rather that be a second PR because this one is already quite big and self-contained.

Reply:
Aah, nice. I hadn't seen that comment yet.

Reply:
I thought about it again and indeed it makes sense to introduce `strict_ordered_ring` simultaneously as the other `strict_ordered_` typeclasses rather with the `ordered_cancel_` ones. So I'm gonna do a bit more in this PR.